### ISSUE TITLE
Vacuum is failing for shared Data Cluster with GCS Storage

### ISSUE THREAD

**Issue by mkan-su** [StarRocks/starrocks#37280]
### Steps to reproduce the behavior (Required)
1. Create a shared data cluster with GCS as storage. Run only FE and CN (have not tried with BE)
2. Create table and insert some data
3. Check CN log (log/cn.INFO)

### Expected behavior (Required)
Vacuum should complete without any error messages

### Real behavior (Required)
Getting below errors and the file is not deleted

```
I1218 12:09:27.028069  4554 vacuum.cpp:88] Deleting staros://11753/log/0000000000002DE9_0000000000000034.log(1/1)
W1218 12:09:27.030179   960 vacuum.cpp:102] Fail to delete: Internal error: starlet err [RequestID=]Delete objects error: Unable to parse Exceptio       nName: NotImplemented Message: A header or query you provided requested a function that is not implemented.

I1218 16:42:52.858620   960 vacuum.cpp:88] Deleting staros://15149/meta/0000000000003B2E_0000000000000001.meta(2/2)
W1218 16:42:52.861732 26832 vacuum.cpp:102] Fail to delete: Internal error: starlet err [RequestID=]Delete objects error: Unable to parse ExceptionName: NotImplemented Message: A header or query you provided requested a function that is not implemented.
```
The above errors are getting constantly repeated. 

### StarRocks version (Required)
3.1.5

**Comment by kevincai** [comment]
confirmed that google gcs doesn't support aws s3 `deleteObjects` interface, need to make change to adapt this compatibility issue.

**Comment by kevincai** [comment]
> Is this issue fixed. We are also facing the same issue !

please watch this issue's status, it will be fixed in next release.

**Comment by kevincai** [comment]
fixed in #37696

**Comment by kevincai** [comment]
done
